File: Sharebook/Client/src/app/core/Public/Models/Interfaces/IMedia.ts
import {IUploadedMediaInfo} from "./DTO/IUploadedMediaInfo";
import {InjectionToken} from "@angular/core";
import { Media } from 'src/app/core/Core/Models/Media';

export interface IMedia {
  name: string;
  data: number[];
  id?: number;
  contentType: string;

  hasBeenUploaded: boolean;
  isReadyToBeUsed: boolean;

  uploadedMediaInfo: IUploadedMediaInfo;

  cloneWithoutMediaInfo(source: IMedia): IMedia;

  cloneWithoutUploadInfo(): IMedia;
}

export const IMediaToken = new InjectionToken<IMedia>('IMedia', {
  providedIn: 'root',
  factory: () => new Media(),
});

File: Sharebook/Server/src/Services/Upload/Sharebook.Upload.API/Models/DTO/IMedia.cs
namespace Sharebook.Upload.Models.DTO
{
    using Sharebook.Upload.DTO;

    public interface IMedia
    {
        string Name { get; set; }

        byte[] Data { get; set; }

        long? Id { get; set; }

        string ContentType { get; set; }

        bool HasBeenUploaded { get; }

        bool IsReadyToBeUsed { get; }

        IUploadedMediaInfo UploadedMediaInfo { get; set; }

        IMedia CloneWithoutMediaInfo(IMedia source);

        IMedia CloneWithoutUploadInfo();
    }
}
